### ISSUE TITLE
Spotube command line control

### ISSUE THREAD

**Issue by beastr45** [KRTirtho/spotube#1177]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

Currently spotube dosent seem to have a way to control it over the command line. I want to make keymaps to show/hide spotube but without documentation on how to interface with spotube with scripts it dosent seem to be possible. To show/hide spotube I have to go to the system tray which is annoying. By default my wm dosent have a system tray so idk how to do it without a system tray. 

### Suggested solution

I would recommend making a very simple program bound to internal spotube functions for use in simple scripting ex. play/pause show/hide skip/fastforward. 

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by BarishNamazov** [comment]
I use `playerctl` for song control, but I would also love to see more CLI features such as getting the next songs in queue, etc. It'd be really helpful to make custom extensions to use Spotube.

**Comment by BarishNamazov** [comment]
> `playerctl` works nicely to control from the CLI, but @BarishNamazov does `playerctl stop` work for you?
@Cody-W-Tucker , checked and I don't see any effect from the stop command. I use the `play-pause` command for my little extension. Are you trying to "reset" the state with the stop command here?

**Comment by Cody-W-Tucker** [comment]
@BarishNamazov yes, I used "Stop" via Waybar to choose a different audio source to control. But only play-pause seems to work with Spotube

```nix
    "custom/media" = {
      format = "{}";
      escape = true;
      interval = 5;
      return-type = "json";
      max-length = 40;
      on-click = "playerctl play-pause";
      on-click-right = "playerctl stop";
      smooth-scrolling-threshold = 1;
      on-scroll-up = "playerctl next";
      on-scroll-down = "playerctl previous";
      exec = "media-player";
    };
```

**Comment by brrock** [comment]
@KRTirtho this would be great

**Comment by brrock** [comment]
Also could u add playlist support and a full headless mode

**Comment by pickpj** [comment]
I would like to add that the volume, position, loop, and shuffle controls in playerctl do not work with spotube, in addition to stop.

**Comment by talon35** [comment]
Was about to open a new issue, but will add to this instead:

❯ playerctl -p spotube pause
❯ playerctl -p spotube play

work fine, but

❯ playerctl -p spotube volume 0.5
❯ playerctl -p spotube volume 0.1-

don't.

I'm on v5.1.0 Flatpak.
I'd really love to see this work!

Thanks

**Comment by Mrjello0** [comment]
yes playerctl volume support would be fantassticc they indeed do not work spotube has been my absolute ideal solution for a low ram alternative to spotify it has everything I want

**Comment by Mrjello0** [comment]
and length functions working would be so amazing too
